enterprise/server/backends/distributed/distributed.go

```diff
@@ -351,8 +351,9 @@ func (c *Cache) Delete(ctx context.Context, d *repb.Digest) error {
 	return status.UnimplementedError("Not yet implemented.")
 }
 
-func (c *Cache) Reader(ctx context.Context, d *repb.Digest, offset int64) (io.Reader, error) {
-	return c.remoteReader(ctx, d, offset)
+func (c *Cache) Reader(ctx context.Context, d *repb.Digest, offset int64) (io.ReadCloser, error) {
+	r, err := c.remoteReader(ctx, d, offset)
```

Comment:
You could just have remoteReader return its ReadCloser directly, and then do away with the autocloser and nopcloser (but you'd have to make sure the callers close these appropriately first!)

Reply:
Went ahead and dumped the autoclose stuff, now everything gets closed in the right place. (Such a big difference to have go-to-reference for that, wow.